rspace/src/main/scala/coop/rchain/rspace/history/RadixTree.scala

```diff
@@ -201,7 +214,13 @@ object RadixTree {
           decodeItem(pos0Next, nodeNext) // Try to decode next item.
         }
 
-      decodeItem(0, emptyNode)
+      try {
+        decodeItem(0, emptyNode)
+      } catch {
+        case _: Exception =>
+          assert(assertion = false, "Error during deserialization: invalid data format")
+          emptyNode
+      }
```

Comment:
We are not using try/catch with Java throw errors, but `ApplicativeError`s inside F context (see `raiseError` in the code).
https://typelevel.org/cats/typeclasses/applicativemonaderror.html

Also we don't want to swallow/hide any error, if error is rethrown, previous error must be supplied. Assert can be use in special cases where error is fatal for the whole running process but it's not recommended.

@hilltracer, @stanislavlyalin Please help Max to convert `decodeItem` function to F value.